### PR TITLE
fix: make valueoptions non-empty

### PR DESCRIPTION
**Pull request by radwouters**
Process changes to closedvariableinterface into these interfaces. This needs to be fixed before we can merge #50 .

### CODE REVIEW

**Comment by SamMousa** [comment]
You must check constructor arguments are not . This PR just removes the default values but still allows bad instances to be constructed.
While adding those checks also make the private variables readonly please!

**Comment by radwouters** [comment]
> You must check constructor arguments are not . This PR just removes the default values but still allows bad instances to be constructed. While adding those checks also make the private variables readonly please!

It does not, right? Since ValueOptions must be non-empty, it does not allow for the construction of bad instances

File: src/Variables/MultipleChoiceVariable.php [hunk 1]
```diff
@@ -18,40 +18,37 @@
...
-         */
-        private array $dataPath,
-        array $rawConfiguration = []
+        private readonly array $dataPath,
+        private readonly array $rawConfiguration = []
     ) {
-        $this->name = $name;
-
+        /* @phpstan-ignore-next-line */
+        assert(count($valueOptions) > 0, throw new InvalidArgumentException('ValueOptions must not be empty'));
```
Comment (SamMousa) [inline src/Variables/MultipleChoiceVariable.php:48]: Why use PHPs `assert`?

> Assertions should be used as a debugging feature only. You may use them for sanity-checks that test for conditions that should always be true and that indicate some programming errors if not or to check for the presence of certain features like extension functions or certain system limits and features.

**Review by SamMousa — CHANGES_REQUESTED** [review]

File: src/Variables/MultipleChoiceVariable.php (hunk 1 above)
Comment (radwouters) [reply src/Variables/MultipleChoiceVariable.php:48]: This is exactly what we discussed, right? Or should just a regular if-statement be used here?

Comment (SamMousa) [reply src/Variables/MultipleChoiceVariable.php:48]: I used the word assertions not referring to the php `assert` function. Static analysers have assertions in the phpdoc.

**Review by SamMousa — APPROVED** [review]

**Pull request merged**